org.lflang/src/org/lflang/generator/LFGenerator.java

```diff
@@ -169,6 +208,49 @@ private GeneratorBase createCppGenerator(FileConfig fileConfig,
         }
     }
 
+    /**
+     * Create a TypeScript code generator
+     *
+     * Since the Typeenerator class is implemented in Kotlin, the class is
+     * not visible from all contexts. If the RCA is run from within Eclipse via
+     * "Run as Eclipse Application", the Kotlin classes are unfortunately not
+     * available at runtime due to bugs in the Eclipse Kotlin plugin. (See
+     * https://stackoverflow.com/questions/68095816/is-ist-possible-to-build-mixed-kotlin-and-java-applications-with-a-recent-eclips)
+     * In this case, the method returns null
+     *
+     * @return A TypeScriptGenerator object if the class can be found
+     */
+    private GeneratorBase createTsGenerator(FileConfig fileConfig,
```

Comment:
`CppGenerator` has a very similar function with duplicated logic. Can we have a generic `createGenerator` method instead?

Reply:
Great suggestion. I merged the functions for CPP and TS. I named it createKotlinGenerator just to differentiate it from other generators. Let me know if this makes sense. :)

Reply:
Yes, makes sense!